apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/style.scss

```
@@ -3,3 +3,14 @@
 		display: none;
 	}
 }
+
+.post-type-page, .post-type-wp_template_part {
+	.editor-styles-wrapper {
+		padding: 0;
+		margin: 0;
+	}
+
+	.template-block {
```

The following will target just the header template block:
```suggestion
		.editor-block-list__block:first-child {
		    .template-block {
			    margin-top: -28px;
		    }
	    }
```
But seems a bit fragile - I wonder it we need to try and get some custom classes in the header and footer blocks like you have done with the content slot in order to more reliably target these?